libs/components/modals/src/lib/modules/modal/modal.component.ts

```diff
@@ -107,30 +91,68 @@ export class SkyModalComponent implements AfterViewInit, OnDestroy {
   public modalHeaderId: string =
     'sky-modal-header-id-' + skyModalUniqueIdentifier.toString();
 
-  public scrollShadow: SkyModalScrollShadowEventArgs;
+  public scrollShadow: SkyModalScrollShadowEventArgs | undefined;
 
   @ViewChild('modalContentWrapper', { read: ElementRef })
-  private modalContentWrapperElement: ElementRef;
+  private modalContentWrapperElement: ElementRef | undefined;
+
+  #hostService: SkyModalHostService;
+  #elRef: ElementRef;
+  #windowRef: SkyAppWindowRef;
+  #componentAdapter: SkyModalComponentAdapterService;
+  #coreAdapter: SkyCoreAdapterService;
+  #dockService: SkyDockService;
+  #mediaQueryService: SkyResizeObserverMediaQueryService | undefined;
+
+  #_ariaRole: string | undefined;
 
   constructor(
-    private hostService: SkyModalHostService,
-    private config: SkyModalConfiguration,
-    private elRef: ElementRef,
-    private windowRef: SkyAppWindowRef,
-    private componentAdapter: SkyModalComponentAdapterService,
-    private coreAdapter: SkyCoreAdapterService,
-    @Host() private dockService: SkyDockService,
+    hostService: SkyModalHostService,
+    config: SkyModalConfiguration,
+    elRef: ElementRef,
+    windowRef: SkyAppWindowRef,
+    componentAdapter: SkyModalComponentAdapterService,
+    coreAdapter: SkyCoreAdapterService,
+    @Host() dockService: SkyDockService,
     @Optional()
-    private mediaQueryService?: SkyResizeObserverMediaQueryService
-  ) {}
+    mediaQueryService?: SkyResizeObserverMediaQueryService
+  ) {
+    this.#hostService = hostService;
+    this.#elRef = elRef;
+    this.#windowRef = windowRef;
+    this.#componentAdapter = componentAdapter;
+    this.#coreAdapter = coreAdapter;
+    this.#dockService = dockService;
+    this.#mediaQueryService = mediaQueryService;
+
+    this.ariaDescribedBy = config.ariaDescribedBy || this.modalContentId;
+    this.ariaLabelledBy = config.ariaLabelledBy || this.modalHeaderId;
+    this.ariaRole = config.ariaRole;
+    this.fullPage = config.fullPage;
+    this.helpKey = config.helpKey;
+    this.tiledBody = config.tiledBody;
+    this.wrapperClass = config.wrapperClass;
+
+    if (this.fullPage) {
+      this.isLargeSize = this.isMediumSize = this.isSmallSize = false;
```

Comment:
Is this line necessary? These properties already have a default value of `false`, so there's no need to set them to `false` again here. Alternatively I think it would be better to eliminate the three discrete properties and have one string union property since only one can ever be true at one time.

```
this.size = config.size?.toLowerCase() || '<whatever the default value is>';
```

Reply:
I made a change to where we have a `size` variable which includes `full-page`. This does change the class declarations in the template some so let me know if you want to chat any more about it.